agent/installer/registry.go

```diff
@@ -32,22 +35,49 @@ func (r *registry) Add(os, k8sVer string, installer osk8sInstaller) {
 	r.osk8sInstallerMap[os][k8sVer] = installer
 }
 
-func (r *registry) ListOS() []string {
+func (r *registry) AddOSAlias(osFilter, os string) {
+	if _, alreadyExist := r.aliasOSMap[osFilter]; alreadyExist {
+                panic(fmt.Sprintf("alias %v already exists", osFilter))
+        }
+
+	r.aliasOSMap[osFilter] = os
+}
+
+func (r *registry) ListOS() ([]string, aliasOSMap) {
 	result := make([]string, 0, len(r.osk8sInstallerMap))
+
 	for os := range(r.osk8sInstallerMap) {
 		result = append(result, os)
 	}
-	return result
+
+	return result, r.aliasOSMap
 }
 
 func (r *registry) ListK8s(os string) []string {
 	result := make([]string, 0, len(r.osk8sInstallerMap[os]))
-	for k8s := range(r.osk8sInstallerMap[os]) {
+	for k8s := range(r.resolveK8sInstallerMap(os)) {
 		result = append(result, k8s)
 	}
 	return result
 }
 
 func (r *registry) GetInstaller(os, k8sVer string) osk8sInstaller {
-	return r.osk8sInstallerMap[os][k8sVer]
+	return r.resolveK8sInstallerMap(os)[k8sVer]
+}
+
+func (r* registry) resolveK8sInstallerMap(os string) k8sInstallerMap {
+	if res, ok := r.osk8sInstallerMap[os]; ok {
+		// There is direct match for the os
+		return res
+	}
+
+	// Try to find os using the alias map
+	for k,v := range r.aliasOSMap {
+		matched, _ := regexp.MatchString(k, os)
+		if matched {
+			return r.osk8sInstallerMap[v]
+		}
+	}
+
+	return nil
```

Comment:
Suggestion: If we remove the aliasOSMap implementation, this can be simplified to something like this
```
func (r *registry) resolveK8sInstallerMap(os string) k8sInstallerMap {
	for k := range r.osk8sInstallerMap {
		var re = regexp.MustCompile(k)
		if re.MatchString(os) {
			return r.osk8sInstallerMap[os]
		}
	}
	return nil
}
```

Reply:
The idea of the map is the same as libraries in linux. In our case the symlink is the osFilter and the bundle is the lib on the right side. The keys of the first map are used for bundle name. Moving them elsewhere will require more refactoring and can be done post FC.

```
lrwxrwxrwx 1 root root      10 юни  4  2020 ld-linux.so.2 -> ld-2.27.so
-rw-r--r-- 1 root root   14024 юни  4  2020 libanl-2.27.so
lrwxrwxrwx 1 root root      14 юни  4  2020 libanl.so.1 -> libanl-2.27.so
-rw-r--r-- 1 root root    5544 юни  4  2020 libBrokenLocale-2.27.so
lrwxrwxrwx 1 root root      23 юни  4  2020 libBrokenLocale.so.1 -> libBrokenLocale-2.27.so
-rwxr-xr-x 1 root root 1926828 юни  4  2020 libc-2.27.so
-rw-r--r-- 1 root root  185920 юни  4  2020 libcidn-2.27.so
lrwxrwxrwx 1 root root      15 юни  4  2020 libcidn.so.1 -> libcidn-2.27.so
-rw-r--r-- 1 root root   38440 юни  4  2020 libcrypt-2.27.so
lrwxrwxrwx 1 root root      16 юни  4  2020 libcrypt.so.1 -> libcrypt-2.27.so

```

Reply:
Thanks for the reference, But do you think there are some advantages to follow the same for our implementation.

Reply:
Yes, we're essentially doing the same thing.